delivery/client.py

```diff
@@ -200,18 +201,31 @@ def update_metadata(
         self,
         data: typing.Iterable[dso.model.ArtefactMetadata],
     ):
-        res = self.session.put(
-            url=self._routes.update_metadata(),
-            json={'entries': [
-                dataclasses.asdict(
-                    artefact_metadata,
-                    dict_factory=ci.util.dict_to_json_factory,
-                ) for artefact_metadata in data
-            ]},
-            timeout=(4, 121),
-        )
+        data_by_types = collections.defaultdict(list)
+
+        for datum in data:
+            data_by_types[datum.meta.type].append(datum)
+
+        processed_types = set()
+        for type, data_for_type in data_by_types.items():
+            res = self.session.put(
+                url=self._routes.update_metadata(),
+                json={'entries': [
+                    dataclasses.asdict(
+                        artefact_metadata,
+                        dict_factory=ci.util.dict_to_json_factory,
+                    ) for artefact_metadata in data_for_type
+                ]},
+                timeout=(4, 121),
+            )
 
-        res.raise_for_status()
+            if not res.ok:
+                logger.warning(
+                    f'error occured during metadata update of {type=}; {processed_types=}'
+                )
+                res.raise_for_status()
```

Comment:
We should not immediately raise upon failed request but rather process the remaining data-types.